text/0003-placeholders-in-conditions.md

```diff
@@ -0,0 +1,145 @@
+# Generalizing template placeholders
+
+## Related issues and PRs
+
+- Reference Issues: https://github.com/cedar-policy/cedar/issues/81
+- Implementation PR(s): 
+
+## Timeline
+
+- Start Date: 2023-06-15
+- Date Entered FCP: 
+- Date Accepted:
+- Date Landed:
+
+# Summary
+
+This RFC proposes to generalize templates in three ways:
+
+1. The `?principal` and `?resource` placeholders may appear in the `when`/`unless` conditions of a policy when they also appear in the policy scope. 
+
+2. Placeholder variables are not limited to `?principal` and `?resource`. A general variable binding mechanism permits introducing additional placeholder variables in the policy, which can appear in the `when`/`unless` conditions. Moreover, these additional variables need not be linked to values of entity type only; any Cedar type is acceptable. Including `?principal` and/or `?resource` in this list is optional; restrictions on their use are enforced either way.
```

Comment:
"this list" what list?

Reply:
Ah, I see later it's the list in `template(...)`. Would be nice to clean up wording here or show an example of the syntax.